### PR TITLE
Added test for returning the policies contained in the policy profiles assigned to a Vm

### PR DESCRIPTION
**Pull request by gtanzillo**
Depends on https://github.com/ManageIQ/manageiq/pull/19930

Test case for -

http://localhost:3000/api/vms/10000000000229/policy_profiles?expand=resources&attributes=miq_policies

```json
{
    "name": "policy_profiles",
    "count": 8,
    "subcount": 1,
    "pages": 1,
    "resources": [
        {
            "href": "http://localhost:3000/api/vms/10000000000229/policy_profiles/10000000000056",
            "id": "10000000000056",
            "name": "GT Profile",
            "description": "GT Profile",
            "set_type": "MiqPolicySet",
            "created_on": "2020-02-21T21:56:30Z",
            "updated_on": "2020-02-21T22:02:42Z",
            "guid": "9369273d-9d21-4be3-b3f2-b29e7fc9188b",
            "read_only": null,
            "set_data": null,
            "mode": "compliance",
            "owner_type": null,
            "owner_id": null,
            "userid": null,
            "group_id": null,
            "miq_policies": [
                {
                    "href": "http://localhost:3000/api/policies/10000000000016",
                    "id": "10000000000016",
                    "name": "d75b2b62-c200-4f9b-ad0c-f3af76834bdb",
                    "description": "GT Compliance Policy",
                    "created_on": "2020-02-20T02:43:03Z",
                    "updated_on": "2020-02-20T02:43:03Z",
                    "expression": null,
                    "towhat": "Vm",
                    "guid": "d75b2b62-c200-4f9b-ad0c-f3af76834bdb",
                    "created_by": "admin",
                    "updated_by": "admin",
                    "notes": null,
                    "active": true,
                    "mode": "compliance",
                    "read_only": null
                },
                {
                    "href": "http://localhost:3000/api/policies/10000000000005",
                    "id": "10000000000005",
                    "name": "8d6e5091-d573-4fec-ae8f-fd7a74291956",
                    "description": "SmartState Analysis",
                    "created_on": "2019-04-23T16:53:57Z",
                    "updated_on": "2019-04-23T16:53:57Z",
                    "expression": null,
                    "towhat": "Vm",
                    "guid": "8d6e5091-d573-4fec-ae8f-fd7a74291956",
                    "created_by": "admin",
                    "updated_by": "admin",
                    "notes": null,
                    "active": true,
                    "mode": "control",
                    "read_only": null
                }
            ]
        }
    ],
    "actions": [
        {
            "name": "assign",
            "method": "post",
            "href": "http://localhost:3000/api/vms/10000000000229/policy_profiles"
        },
        {
            "name": "unassign",
            "method": "post",
            "href": "http://localhost:3000/api/vms/10000000000229/policy_profiles"
        },
        {
            "name": "resolve",
            "method": "post",
            "href": "http://localhost:3000/api/vms/10000000000229/policy_profiles"
        }
    ],
    "links": {
        "self": "http://localhost:3000/api/vms/10000000000229/policy_profiles?expand=resources&attributes=miq_policies&offset=0",
        "first": "http://localhost:3000/api/vms/10000000000229/policy_profiles?expand=resources&attributes=miq_policies&offset=0",
        "last": "http://localhost:3000/api/vms/10000000000229/policy_profiles?expand=resources&attributes=miq_policies&offset=0"
    }
}
```
Fixes https://github.com/ManageIQ/manageiq-api/issues/745

### CODE REVIEW

File: spec/requests/policies_spec.rb [hunk 1]
```diff
@@ -98,6 +98,36 @@ def test_policy_profile_query(object, api_object_policy_profiles_url)
...
+    #       "miq_policies" => a_collection_including(
+    #         a_hash_including(
+    #           "guid" => Array.wrap([p1.guid, p2.guid])
+    #         )
+    #       )
+    #     )
+    #   )
+    # )
+    # expect(response.parsed_body).to include(expected)
+    # So I instead went with this -
```
Comment (abellotti) [inline spec/requests/policies_spec.rb]: The following seems to work, give it a whirl ....

```
    expected = {
      "name"  => "policy_profiles",
      "count" => 2,
      "resources" => a_collection_containing_exactly(
        a_hash_including(
          "guid" => ps1.guid,
          "miq_policies" => a_collection_containing_exactly(
            a_hash_including("guid" => p1.guid),
            a_hash_including("guid" => p2.guid)
          )
        ),
        a_hash_including(
          "guid" => ps2.guid,
          "miq_policies" => a_collection_containing_exactly(
            a_hash_including("guid" => p3.guid)
          )
        )
      )
    }
    expect(response.parsed_body).to include(expected)
```

Comment (gtanzillo) [reply spec/requests/policies_spec.rb]: Thanks @abellotti that works!

Comment (abellotti) [reply spec/requests/policies_spec.rb]: No problem.  Maybe just update the alignment there for Mr. Rubocop 🤖  Thanks.

**Comment by abellotti** [comment]
LGTM!! Thanks @gtanzillo for the API Tests 👍  just the rubocop here and dependent PR merged, and we should be good to go.

**Comment by Fryguy** [comment]
I know that tests are in manageiq-api, but we really should have tests here as well, because if someone changes it you end up in a "core is green, but api is red" case.

**Comment by Fryguy** [comment]
Dependent PR was merged, so bouncing this.

**Comment by abellotti** [comment]
Yeay, ✅ Travis CI -  merging.  Thanks @gtanzillo for the enhancement and added spec. 👍

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 39a4e80efee27736e28d06b428719ee49021a09a
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Thu Apr 9 17:38:41 2020 -0400

    Merge pull request #766 from gtanzillo/test-has_many-for-set

    Added test for returning the policies contained in the policy profiles assigned to a Vm

    (cherry picked from commit 0f93208c120442a36e6e8a4bee198bad13842596)
```

**Pull request merged**